client/my-sites/design-preview/index.js

```diff
@@ -165,24 +167,39 @@ const DesignPreview = React.createClass( {
 			return;
 		}
 		event.preventDefault();
-		// TODO: if the href is on the current site, load the href as a preview and fetch markup for that url
 	},
 
-	getPreviewUrl( site ) {
-		if ( ! site ) {
+	getParsedPreviewUrl() {
+		if ( ! this.props.selectedSiteUrl && ! this.props.previewUrl ) {
 			return null;
 		}
+		const baseUrl = this.props.previewUrl ? this.props.previewUrl : this.props.selectedSiteUrl;
+		const parsed = url.parse( baseUrl, true );
+		delete parsed.search;
+		return parsed;
+	},
 
-		const parsed = url.parse( site.options.unmapped_url, true );
+	getPreviewUrl() {
+		const parsed = this.getParsedPreviewUrl();
+		if ( ! parsed ) {
+			return null;
+		}
 		parsed.query.iframe = true;
 		parsed.query.theme_preview = true;
-		if ( site.options && site.options.frame_nonce ) {
-			parsed.query['frame-nonce'] = site.options.frame_nonce;
+		if ( this.props.selectedSiteNonce ) {
+			parsed.query[ 'frame-nonce' ] = this.props.selectedSiteNonce;
 		}
-		delete parsed.search;
 		return url.format( parsed ) + '&' + this.state.previewCount;
 	},
 
+	getExternalUrl() {
+		const parsed = this.getParsedPreviewUrl();
```

Comment:
I guess that's a good point. It was more helpful when figuring out the preview URL was more complex, but now... not so much.